### PR TITLE
feat(prejoin): added env and launch method

### PR DESCRIPTION
**Pull request by shnaaz**
# COMPLETES #< INSERT LINK TO ISSUE >
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-458895
## This pull request addresses

< DESCRIBE THE CONTEXT OF THE ISSUE >

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts [hunk 1]
```diff
@@ -160,6 +164,7 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
           os: getOSNameInternal(),
           browser: getBrowserName(),
           browserVersion: getBrowserVersion(),
+          clientLaunchMethod: options?.clientLaunchMethod, // do we want something by default
```
Comment (torgeadelin) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:167]: i think we should do it at the end of the function, 
if options. clientLaunchMethod {
  origin. clientLaunchMethod = ...
}

File: packages/@webex/internal-plugin-metrics/src/metrics.types.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export type SubmitClientEventOptions = {
   showToUser?: boolean;
   correlationId?: string;
   preLoginId?: string;
+  environment?: Event['origin']['environment'];
```
Comment (torgeadelin) [inline packages/@webex/internal-plugin-metrics/src/metrics.types.ts:18]: use the new types u added

File: packages/@webex/internal-plugin-metrics/src/metrics.types.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@ export type SubmitClientEventOptions = {
...
+  clientLaunchMethod?: Event['origin']['clientInfo']['clientLaunchMethod'];
```
Comment (torgeadelin) [inline packages/@webex/internal-plugin-metrics/src/metrics.types.ts:19]: same here

**Review by torgeadelin — CHANGES_REQUESTED** [review]

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts (hunk 1 above)
Comment (shnaaz) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:167]: But clientLaunchMethod is at origin.clientInfo.clientLaunchMethod level so I think you mean at end of the function set it at that level

Comment (torgeadelin) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts]: yes sorry

File: packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts [hunk 4]
```diff
@@ -160,6 +164,7 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
           os: getOSNameInternal(),
           browser: getBrowserName(),
           browserVersion: getBrowserVersion(),
+          clientLaunchMethod: options?.clientLaunchMethod,
```
Comment (torgeadelin) [inline packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:167]: remove this

Comment (shnaaz) [reply packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts:167]: oops!

**Review by torgeadelin — APPROVED** [review]

**Review by Coread — APPROVED** [review]

**Pull request merged**